sw/device/tests/sim_dv/uart_tx_rx_test.c

```diff
@@ -491,6 +504,36 @@ static bool execute_test(const dif_uart_t *uart) {
   return true;
 }
 
+void config_external_clock(void) {
+  uint32_t extclk_ctrl_reg = 0;
+  dif_lc_ctrl_t lc;
+  dif_clkmgr_t clkmgr;
+  mmio_region_t lc_reg = mmio_region_from_addr(TOP_EARLGREY_LC_CTRL_BASE_ADDR);
+  mmio_region_t clkmgr_reg =
+      mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR);
+
+  CHECK_DIF_OK(dif_lc_ctrl_init(lc_reg, &lc));
+  CHECK_DIF_OK(dif_clkmgr_init(clkmgr_reg, &clkmgr));
+
+  LOG_INFO("Read and check LC state.");
+  dif_lc_ctrl_state_t curr_state;
+  CHECK_DIF_OK(dif_lc_ctrl_get_state(&lc, &curr_state));
+  CHECK(curr_state == kDifLcCtrlStateRma,
+        "LC State isn't in kDifLcCtrlStateRma!");
+
+  // Enable external clock
+  LOG_INFO("Configure clkmgr to enable external clock");
+  extclk_ctrl_reg = bitfield_field32_write(
+      extclk_ctrl_reg, CLKMGR_EXTCLK_CTRL_SEL_FIELD, kMultiBitBool4True);
+  extclk_ctrl_reg = bitfield_field32_write(
+      extclk_ctrl_reg, CLKMGR_EXTCLK_CTRL_LOW_SPEED_SEL_FIELD,
+      kUseLowSpeedSel ? kMultiBitBool4True : kMultiBitBool4False);
+  mmio_region_write32(clkmgr_reg, CLKMGR_EXTCLK_CTRL_REG_OFFSET,
+                      extclk_ctrl_reg);
+
+  LOG_INFO("External clock setting completes");
+}
```

Comment:
Shouldn't this be a DIF? If one does not exist, could you create on by simply moving this functionality to a new clkmgr DIF?

Reply:
Good point. I moved it to clkmgr DIF. PTAL